lib/cluster/cluster.c

```diff
@@ -119,6 +119,38 @@ crm_cluster_disconnect(crm_cluster_t *cluster)
     }
 }
 
+/*!
+ * \brief Allocate a new \p crm_cluster_t object
+ *
+ * \return A newly allocated \p crm_cluster_t object (guaranteed not \p NULL)
+ * \note The caller is responsible for freeing the return value using
+ *       \p pcmk_cluster_free().
+ */
+crm_cluster_t *
+pcmk_cluster_new(void)
+{
+    crm_cluster_t *cluster = calloc(1, sizeof(crm_cluster_t));
+
+    CRM_ASSERT(cluster != NULL);
+    return cluster;
+}
+
+/*!
+ * \brief Free a \p crm_cluster_t object and its dynamically allocated members
+ *
+ * \param[in,out] cluster  Cluster object to free
+ */
+void
+pcmk_cluster_free(crm_cluster_t *cluster)
+{
+    if (cluster == NULL) {
+        return;
+    }
+    free(cluster->uuid);
+    free(cluster->uname);
+    free(cluster);
+}
```

Comment:
FYI any existing static declarations of crm_cluster_t should be replaced with the new functions. No need for a separate free function for the members only. Basically we're deprecating static declarations and C library allocations of crm_cluster_t.

Reply:
Well this is growing... about to add dynamic `crm_cluster_t` to the CIB manager. Looks like the `cib_cleanup()` function never gets called anywhere lol

Reply:
`pacemaker-based.c` looks like the only static, non-pointer declaration though